core/src/ldml/ldml_processor.cpp

```diff
@@ -168,21 +222,83 @@ ldml_processor::process_event(
 
     switch (vk) {
     case KM_KBP_VKEY_BKSP:
-      state->context().pop_back();
-      state->actions().push_backspace(KM_KBP_BT_UNKNOWN); // Assuming we don't know the character
+      {
+        KMX_DWORD last_char = 0UL; // Assuming we don't know the character
+        // attempt to get the last char
+        auto end = state->context().rbegin();
+        if(end != state->context().rend()) {
+          if((*end).type == KM_KBP_CT_CHAR) {
+            last_char = (*end).character;
+          }
+        }
+        if (last_char == 0UL) {
+          state->actions().push_backspace(KM_KBP_BT_UNKNOWN);
+        } else {
+          state->actions().push_backspace(KM_KBP_BT_CHAR, last_char);
+        }
+        state->context().pop_back();
+      }
       break;
     default:
+      // from kmx_processor.cpp
+      // Construct a context buffer from the items up until the last KM_KBP_CT_MARKER marker
+      ldml::string_list ctxt;
+      auto cp = state->context();
+      // We're only interested in as much of the context as is a KM_KBP_CT_CHAR.
+      // This will stop at the KM_KBP_CT_MARKER type.
```

Comment:
Shouldn't this be `KM_KBP_BT_MARKER` and `KM_KBP_BT_CHAR`? (Same a few lines above)